plugins/inputs/kubernetes/kubernetes.go

```diff
@@ -101,10 +107,17 @@ func (k *Kubernetes) gatherSummary(baseURL string, acc telegraf.Accumulator) err
 	}
 
 	if k.RoundTripper == nil {
+		// Set default values
+		if k.ResponseTimeout.Duration < time.Second {
+			k.ResponseTimeout.Duration = time.Second * 5
+		}
+		if err != nil {
+			return err
+		}
 		k.RoundTripper = &http.Transport{
 			TLSHandshakeTimeout:   5 * time.Second,
 			TLSClientConfig:       tlsCfg,
-			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+			ResponseHeaderTimeout: k.ResponseTimeout.Duration,
```

Comment:
Just thinking aloud, we should switch to a http.client sometime so we can include the time to get the body.  We are no worse off now with this change so not required.